mypy/main.py

```diff
@@ -129,6 +128,9 @@ def flush_errors(new_messages: List[str], serious: bool) -> None:
     elif code:
         sys.exit(code)
 
+    # HACK: keep res alive so that mypyc won't free it before the hard_exit
+    res2 = res  # noqa
```

Comment:
If mypyc would be a bit smarter, this could be optimized away. What about doing something a bit more involved like `list([res])` which likely would not get optimized away even by a clever compiler?